book/src/proptest/state-machine.md

```diff
@@ -0,0 +1,174 @@
+# State Machine testing
+
+## When to use State Machine testing?
+
+State machine testing automates the checking of properties of a system under test (SUT) against an abstract reference state machine definition. It does this by trying to discover a counter-example that breaks the defined properties of the system and attempts to shrink it to a minimal sequence of transitions that still reproduce the issue.
+
+State machines are a very useful abstraction for reasoning about code. Many things from low-level to high-level logic and anywhere in between can be modelled as a state machine. They are very effective for modelling effectful code, that is code that performs some state changes that can be too hard to test thoroughly with a more manual approach or too complex to verify formally.
+
+Some fitting examples to give you an idea include (by no means exhaustive):
+
+- A data structure with an API that mutates its state
+- An API for a database
+- Interactions between a client(s) and a server
+
+There is some initial investment needed to set the test up and it usually takes a bit more time to run than simple prop tests, but if correctness is important for your use case, you'll be rewarded with a test that is so effective at discovering bugs it might feel almost magical, but as you'll see, [you could have easily implemented it yourself](#how-does-it-work). Also, once you have the test setup, it is much easier to extend it and add new properties to check.
+
+## How to use it
+
+Before using state machine testing, it is recommended to be at least familiar with the basic concepts of Proptest itself as it's built on its essential foundations. That is:
+
+- Strategies are composed and used to generate inputs to a state machine test.
+- A test will attempt to shrink the generated transitions sequence on a discovery of a case that breaks some properties.
+- It will capture regressions file with a seed that can be used to deterministically repeat the found case.
+
+In short, use `ReferenceStateMachine` and `StateMachineTest` to implement your state machine test and `prop_state_machine!` macro to run it.
```

Comment:
Thanks for pointing this out, though perhaps the issue is in the wording of the bullet points themselves. I'll try to make those clearer, because

> - Strategies are composed and used to generate inputs to a state machine test.

These are not state machine specific strategies - they are composed from regular proptest's foundations

> - A test will attempt to shrink the generated transitions sequence on a discovery of a case that breaks some properties.

The sequence of transitions is a strategy and like other strategies it does shrink to a minimal case that still reproduces the issue